src/data/onChainPoolDataViaReadContract.ts

```diff
@@ -18,24 +30,22 @@ type Result =
 
 type Results = Result[];
 
-const abi = parseAbi([
-    'function getPoolTokens(bytes32 poolId) view returns (address[] tokens, uint256 lastChangeBlock)',
-    'function getSwapFeePercentage() view returns (uint256)',
-    'function percentFee() view returns (uint256)',
-    'function protocolPercentFee() view returns (uint256)',
-    'function getNormalizedWeights() view returns (uint256[])',
-    'function totalSupply() view returns (uint256)',
-    'function getVirtualSupply() view returns (uint256)',
-    'function getActualSupply() view returns (uint256)',
-    'function getTargets() view returns (uint256 lowerTarget, uint256 upperTarget)',
-    'function getTokenRates() view returns (uint256, uint256)',
-    'function getWrappedTokenRate() view returns (uint256)',
-    'function getAmplificationParameter() view returns (uint256 value, bool isUpdating, uint256 precision)',
-    'function getPausedState() view returns (bool)',
-    'function inRecoveryMode() view returns (bool)',
-    'function getRate() view returns (uint256)',
-    'function getScalingFactors() view returns (uint256[] memory)',
-]);
+const abis = [
+    ...ComposableStablePoolV5Abi,
+    ...ConfigurableRightsPoolAbi,
+    ...ConvergentCurvePoolAbi,
+    ...FxPoolAbi,
+    ...GyroEV2Abi,
+    ...LiquidityBootstrappingPoolAbi,
+    ...ManagedPoolAbi,
+    ...MetaStablePoolAbi,
+    ...PhantomStablePoolAbi,
+    ...WeightedPoolAbi,
+    ...VaultAbi
+]
+
+const uniqueAbiElements = new Map(abis.map(abi => [JSON.stringify(abi), abi]));
```

Comment:
I'm sorry, but I just realized the existing abis are not `.json`, but `.ts` files exported as `const`.
I think we should benefit from keeping the same standard. 
Can you please update your implementation to that instead?

Reply:
I have moved the abis to be exported from .ts files.